src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -86,4 +87,31 @@ private static void setCheckName(InputConfiguration.Builder inputConfigBuilder,
         final String checkPath = getPackageFromFilePath(filePath) + "." + checkName + "Check";
         inputConfigBuilder.setCheckName(checkPath);
     }
+
+    private static void setCheckAttributes(InputConfiguration.Builder inputConfigBuilder,
+                                           String filePath, List<String> lines)
+                    throws CheckstyleException {
+        int lineNo = 2;
+        for (String line = lines.get(lineNo); !line.contains("*/"); line = lines.get(++lineNo)) {
+            if (line.isEmpty()) {
+                continue;
+            }
+            try {
+                final int equalPosition = line.indexOf('=');
+                final String attribute = line.substring(0, equalPosition - 1);
+                if (line.contains("(default)")) {
+                    final String value = line.substring(line.indexOf(')') + 1);
+                    inputConfigBuilder.addDefaultAttribute(attribute, value);
+                }
+                else {
+                    final String value = line.substring(equalPosition + 2);
+                    inputConfigBuilder.addNonDefaultAttribute(attribute, value);
+                }
+            }
+            catch (IndexOutOfBoundsException ex) {
+                throw new CheckstyleException("Config not formatted properly in file "
```

Comment:
I am not sure we should throw exception, since this functionality is for test purposes only and regular users shouldn't see it

Reply:
it is ok to throw exception, we just need execution failure, exception is good here.